test/mapmatch.cc

```diff
@@ -912,35 +949,78 @@ void test_now_matches() {
+      R"([{"lat": 52.0826293, "lon": 5.1267623, "type": "break", "node_snap_tolerance":0},
+          {"lat": 52.0835867, "lon": 5.1276355, "type": "break", "node_snap_tolerance":0},
+          {"lat": 52.0837127, "lon": 5.1277763, "type": "break", "node_snap_tolerance":0},
+          {"lat": 52.0839615, "lon": 5.1280204, "type": "break", "node_snap_tolerance":0},
+          {"lat": 52.0841756, "lon": 5.1282906, "type": "break", "node_snap_tolerance":0}]})",
+      R"([{"lat": 52.0609108, "lon": 5.0924059, "type": "break", "node_snap_tolerance":0},
+          {"lat": 52.0605926, "lon": 5.0962937, "type": "break", "node_snap_tolerance":0},
+          {"lat": 52.0604866, "lon": 5.0975675, "type": "break", "node_snap_tolerance":0},
+          {"lat": 52.0601766, "lon": 5.1005663, "type": "break", "node_snap_tolerance":0}]})",
+  };
+
+  // TODO: not enough tests, add more. specifically add some with discontinuities and then run two
+  // routes for those
+
+  api_tester tester;
+  for (const auto& test_case : test_cases) {
+    printf("\nroute:\n");
+    // do the same thing for the route api
+    auto route_test_case = R"({"costing":"auto","locations":)" + test_case;
+    auto route_api = tester.route(route_test_case);
+    std::vector<std::vector<std::pair<double, std::string>>> route_times;
+    for (const auto& route : route_api.trip().routes()) {
+      printf("leg:\n");
+      route_times.emplace_back();
+      for (const auto& leg : route.legs()) {
+        route_times.back().emplace_back(leg.node().rbegin()->elapsed_time(), leg.shape());
+        printf("%.2f %s\n", route_times.back().back().first,
+               route_times.back().back().second.c_str());
+      }
+    }
+
+    printf("match:\n");
+    // do the map match for this test and record each routes legs times
+    auto match_test_case = R"({"costing":"auto", "shape_match":"map_snap","shape":)" + test_case;
+    auto match_api = tester.match(match_test_case);
+    std::vector<std::vector<std::pair<double, std::string>>> match_times;
+    for (const auto& route : match_api.trip().routes()) {
+      printf("leg:\n");
+      match_times.emplace_back();
+      for (const auto& leg : route.legs()) {
+        match_times.back().emplace_back(leg.node().rbegin()->elapsed_time(), leg.shape());
+        printf("%.2f %s\n", match_times.back().back().first,
+               match_times.back().back().second.c_str());
+      }
+    }
+
+    // they should not disagree (unless the map match is very vague)
+    if (match_times.size() != route_times.size())
+      throw std::logic_error("Number of routes differs");
+    for (size_t i = 0; i < match_times.size(); ++i) {
+      if (match_times[i].size() != route_times[i].size())
+        throw std::logic_error("Number of legs differs");
+      for (size_t j = 0; j < match_times[i].size(); ++j) {
+        if (!valhalla::midgard::equal(match_times[i][j].first, route_times[i][j].first, 0.1))
```

Comment:
had to add an epsilon of 100 milliseconds because of floating point error accumulation.